frontend/src/state/backendApi.ts

```diff
@@ -741,7 +741,12 @@ const apiStore = {
         const rq = cachedApiRequest(`./api/schemas/subjects/${subjectName}/versions/${version}`, force) as Promise<SchemaDetailsResponse>;
 
         return rq
-            .then(({ schemaDetails }) => (this.schemaDetails = schemaDetails))
+            .then(({ schemaDetails }) => { 
+              if ( schemaDetails && typeof schemaDetails.schema === "string" && schemaDetails.type !== "PROTOBUF") {
```

Comment:
I think the typeof check is not really needed as we rely on the correctness of our typescript types, but @rikimaru0345 might chime in here as he's mostly responsible for the frontend code.

Reply:
@weeco The main reason I did this was that even for Avro schemas it was not showing the detail in the schema details page, and the issue was because the schema object is sent as string from the backend, adding this validation/conversion solved the issue and showed the schema correctly,

Reply:
no, the `typeof` check is fine.
yes, it is assumed that the backend currently sends `schema` as a string and not anything else, but defensive programming is always a good idea.